### DISCUSSION TITLE
How do I manually configure my rust formatter?

### DISCUSSION THREAD

**Discussion by Dangerise** [helix-editor/helix#9521] [Q&A]
I need to configure my rust formatter manually for some reasons, I try to simply use rustfmt, but rustfmt will write nothing to stdout when the input is not correct, and it make my source file empty.

So, how do I avoid that

```
formatter = { command = "bash", args = ["-c", "rustfmt | dx fmt -f -"] }
```

**Comment by kirawi** [comment]
It needs to have a non-zero return to indicate that formatting failed IIRC

**Reply by Dangerise** [reply]
I learnt, so is there a way to do this without editing the source code?

**Reply by univerz** [reply]
> I learnt, so is there a way to do this without editing the source code?

did you solve it?
